host_modules/docker_service.py

```diff
@@ -0,0 +1,86 @@
+"""Docker service handler"""
+
+from host_modules import host_service
+import docker
+import signal
+import errno
+
+MOD_NAME = "docker_service"
+
+
+class DockerService(host_service.HostModule):
```

Comment:
>DockerService [](http://example.com/codeflow?start=6&length=13)

Could you reuse the code inside the SystemdService class?

Reply:
We could do that. But eventually we still need a docker service for upgrade because even if we can stop the old container using systemd, launching new container with new image:tag still requires a `docker run` command (correct me if I am wrong) which still requires this `docker service`. 

If that's so we might as well let this relatively code in and build on top of it.

Reply:
Agree with docker service vs systemd service. I feel there is value in ALLOWED_SERVICES. Could you reuse that array or implemnet a similar ALLOWED_CONTAINERS.

Reply:
Done.